forest/src/main.rs

```diff
@@ -65,6 +66,15 @@ fn main() {
         }
     };
 
+    // Read Genesis file
+    let genesis_buffer: Option<BufReader<File>> = match config.genesis_file.clone() {
+        Some(path) => {
+            let file = File::open(path).expect("Could not load genesis");
+            Some(BufReader::new(file))
+        }
+        None => None,
```

Comment:
Should the reading of the genesis be here? Intuitively I would think that the parsing of genesis should not happen in the ChainSync process? Thoughts?